presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -255,6 +255,14 @@ public static ConnectorPageSource createParquetPageSource(
             for (BlockMetaData block : footerBlocks.build()) {
                 if (predicateMatches(parquetPredicate, block, finalDataSource, descriptorsByPath, parquetTupleDomain, failOnCorruptedParquetStatistics)) {
                     blocks.add(block);
+                    hiveFileContext.incrementCounter("parquet.blocksRead", 1);
```

Comment:
FYI, for each counter, we also track its count, min, max, sum.
So in your use case, the count of the rowsRead and totalBytesRead are actually the sum of this blocksRead. So you could remove this parquet.blocksRead as it is not carrying more information. But if you want to keep it for better visualization, that is also fine.
Same applies for the parquet.blocksSkipped below.

Reply:
Thanks for the tip. I've decided to remove blocksRead/Skipped since row and totalByte information is probably more important than the number of blocks. Because of this, it'll probably be fine to just get the block count from the counter for rows or totalBytes.